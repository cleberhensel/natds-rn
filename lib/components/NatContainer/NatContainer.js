import React from 'react';
import PropTypes from 'prop-types';

import { View } from './NatContainer.styles';

export const NatContainer = ({ children, ...props }) => (
  <View {...props}>
    { children }
  </View>
);

NatContainer.propTypes = {
  children: PropTypes.node.isRequired,
  marginBottom: PropTypes.string,
  marginLeft: PropTypes.string,
  marginRight: PropTypes.string,
  marginTop: PropTypes.string,
  paddingBottom: PropTypes.string,
  paddingLeft: PropTypes.string,
  paddingRight: PropTypes.string,
  paddingTop: PropTypes.string,
};

NatContainer.defaultProps = {
  marginBottom: 'none',
  marginLeft: 'none',
  marginRight: 'none',
  marginTop: 'none',
  paddingBottom: 'small',
  paddingLeft: 'small',
  paddingRight: 'small',
  paddingTop: 'small',
};
