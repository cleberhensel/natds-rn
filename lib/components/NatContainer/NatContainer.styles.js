import styled from 'styled-components/native';
import { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';
import { sizes } from '../../tokens/sizes';

export const View = styled.View`
  margin-top: ${props => sizes[props.marginTop]}px;
  margin-bottom: ${props => sizes[props.marginBottom]}px;
  margin-right: ${props => sizes[props.marginRight]}px;
  margin-left: ${props => sizes[props.marginLeft]}px;
  padding-top: ${props => sizes[props.paddingTop]}px;
  padding-bottom: ${props => sizes[props.paddingBottom]}px;
  padding-right: ${props => sizes[props.paddingRight]}px;
  padding-left: ${props => sizes[props.paddingLeft]}px;
  ${props => props.center && 'justify-content: center; align-items: center;'}
  background-color: ${props => (props.color ? props.color : ThemeNaturaLight.palette.background.paper)}
`;
