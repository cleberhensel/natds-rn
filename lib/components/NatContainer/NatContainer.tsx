import React from "react";
import { View } from "./NatContainer.styles";
type NatContainerProps = {
  marginBottom?: string,
  marginLeft?: string,
  marginRight?: string,
  marginTop?: string,
  paddingBottom?: string,
  paddingLeft?: string,
  paddingRight?: string,
  paddingTop?: string
};
export const NatContainer: React.SFC<NatContainerProps> = ({
  children,
  ...props
}) => <View {...props}>{children}</View>;
NatContainer.defaultProps = {
  marginBottom: "none",
  marginLeft: "none",
  marginRight: "none",
  marginTop: "none",
  paddingBottom: "small",
  paddingLeft: "small",
  paddingRight: "small",
  paddingTop: "small"
};
