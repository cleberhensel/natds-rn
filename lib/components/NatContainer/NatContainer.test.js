import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import 'jest-styled-components/native';

import { NatContainer } from './NatContainer';

describe('Container component', () => {
  it('Should render container component correctly', () => {
    const container = renderer.create(
      <NatContainer>
        <Text>Hello</Text>
      </NatContainer>,
    ).toJSON();

    expect(container).toMatchSnapshot();
  });

  it('Should render container component with default styling', () => {
    const container = renderer.create(
      <NatContainer>
        <Text>Hello</Text>
      </NatContainer>,
    ).toJSON();

    expect(container).toHaveStyleRule('background-color', '#FFFFFF');
  });

  it('Should render container component with custom styling', () => {
    const container = renderer.create(
      <NatContainer color="red" center>
        <Text>Hello</Text>
      </NatContainer>,
    ).toJSON();

    expect(container).toHaveStyleRule('background-color', 'red');
    expect(container).toHaveStyleRule('justify-content', 'center');
    expect(container).toHaveStyleRule('align-items', 'center');
  });

  it('should pass the props center', () => {
    const containerCenter = renderer.create(
      <NatContainer center>
        <Text>Hello</Text>
      </NatContainer>,
    ).toJSON();

    expect(containerCenter).toHaveStyleRule('justify-content', 'center');
    expect(containerCenter).toHaveStyleRule('align-items', 'center');
  });
});
