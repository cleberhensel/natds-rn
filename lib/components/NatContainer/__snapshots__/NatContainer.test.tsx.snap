// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatContainer component Should render container component correctly: NatContainer snapshot 1`] = `
<View
  marginBottom="none"
  marginLeft="none"
  marginRight="none"
  marginTop="none"
  paddingBottom="small"
  paddingLeft="small"
  paddingRight="small"
  paddingTop="small"
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
